src/videoplayer.tsx

```diff
@@ -0,0 +1,222 @@
+        type: "application/x-mpegurl",
+        lowLatency: false,
+      },
+    ],
+    // Poster image
+    poster: props.posterUrl,
+  };
+
+  player.preload = "auto";
+
+  return player;
+};
+
+const buildApi = (player): IVideoplayerApi => {
+  return {
+    seek: (time: number, play?: boolean) => {
+      if (!player) return;
+
+      player.currentTime = time;
+      if (play) player.play();
+    },
+    play: () => {
+      player?.play();
+    },
+    pause: () => {
+      player?.pause();
+    },
+    getCurrentTime: () => {
+      return player?.currentTime ?? 0;
+    },
+    getCurrentFrame: () => {
+      const activeQuality = player?.videoTracks[0]?.activeQuality;
+      if (!activeQuality) return null;
+
+      return Math.round(
+        (player.currentTime - activeQuality.firstFrame) *
+          activeQuality.frameRate -
+          0.5
+      );
+    },
+    seekToFrame: (targetFrame) => {
+      const activeQuality = player?.videoTracks[0]?.activeQuality;
+      if (!activeQuality) return;
+
+      const time =
+        (targetFrame + 0.5) / activeQuality.frameRate +
+        activeQuality.firstFrame;
+      player.currentTime = time;
+    },
+  };
+};
+
+const defaultProps: Partial<IVideoplayerProps> = {
+  autoplay: true,
+};
+
+const preAuth = async (identifier, videoUrl) => {
+  if (platform.name !== "Safari") return;
+
+  const { auth, url } = videoUrl;
+  if (!auth) return;
+
+  const uri = new URL(url);
+  const endpoint = `${uri.protocol}//${uri.hostname}/api/v2/cas/client/preauth/${identifier}`;
+
+  try {
+    await request.post(endpoint).send(auth);
+  } catch {}
+};
+
+const callback = (cb, ...args) => {
+  if (!cb) return;
+
+  try {
+    cb(...args);
+  } catch (e) {
+    console.warn("videoplayer callback error", e);
+  }
+};
+
+export const Videoplayer: React.FC<IVideoplayerProps> = React.forwardRef(
+  (props, ref) => {
+    props = { ...defaultProps, ...props };
+
+    const identifier = buildId();
+    const isScriptLoaded = useLoadScript(THEO_SCRIPT_URL);
+    const isCssLoaded = useLoadCss(THEO_CSS_URL);
+    useLoadScript(CHROMECAST_SCRIPT_URL);
+
+    const elementRef = useRef();
+
+    useIEGuard();
+    const videoUrl = useVideoUrl(props.video);
+
+    const [player, setPlayer] = useState(null);
+
+    useEffect(() => {
+      if (!isScriptLoaded) return;
+      if (!videoUrl) return;
+
+      const instance = buildPlayer(props, videoUrl, elementRef);
+
+      (async () => {
+        await preAuth(identifier, videoUrl);
+        setPlayer(instance); // player can be used from now on
+      })();
+
+      return () => {
+        instance?.destroy();
+      };
+    }, [videoUrl, isScriptLoaded]);
+
+    useAnalytics(player, videoUrl, identifier);
+
+    useEffect(() => {
+      if (!player) return;
+
+      if (props.autoplay) player.play();
+
+      callback(props.onInitialized, buildApi(player));
+
+      player.addEventListener("play", (evt) => callback(props.onPlay, evt));
```

Comment:
Would this be more readable?
```suggestion
      player.addEventListener("play", (evt) => callback(() => props.onPlay(evt)));
```

Reply:
Alternatively it could be as simple as that, with a small modification of `callback` 
```suggestion
      player.addEventListener("play", callback(props.onPlay));
```